test/test_rest_vol.c

```diff
@@ -2963,28 +2957,26 @@ test_create_attribute_on_datatype(void)
 
     H5E_BEGIN_TRY
     {
-#if 0
         if (H5Aclose(attr_id) < 0)
             TEST_ERROR
         if (H5Aclose(attr_id2) < 0)
             TEST_ERROR
-#endif
 
 #ifdef RV_CONNECTOR_DEBUG
         puts("Attempting to open the attributes with H5Aopen_by_idx\n");
 #endif
 
-        if ((/*attr_id = */ H5Aopen_by_idx(type_id, "/" ATTRIBUTE_CREATE_ON_DATATYPE_DTYPE_NAME,
-                                           H5_INDEX_NAME, H5_ITER_INC, 0, H5P_DEFAULT, H5P_DEFAULT)) >= 0) {
+        if ((attr_id =
+                 H5Aopen_by_idx(type_id, ".", H5_INDEX_NAME, H5_ITER_INC, 0, H5P_DEFAULT, H5P_DEFAULT)) < 0) {
```

Comment:
These tests were intentionally using an absolute path to try and catch any issues there, but as a general note, you should try this PR with the API tests in the library or the external VOL tests repo since they do a better job of testing the by_index functions.

Reply:
I've been using the external repo for testing, this is just to keep the CI tests in github happy.